clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -42,17 +43,32 @@
      * Create a new cluster with the given nodes and partitions
      * @param nodes The nodes in the cluster
      * @param partitions Information about a subset of the topic-partitions this cluster hosts
+     * @deprecated Use the Cluster constructor with 4 parameters
      */
+    @Deprecated
     public Cluster(Collection<Node> nodes,
                    Collection<PartitionInfo> partitions,
                    Set<String> unauthorizedTopics) {
-        this(false, nodes, partitions, unauthorizedTopics);
+        this(false, nodes, partitions, unauthorizedTopics, Collections.<String>emptySet());
```

Comment:
Interesting point. I had just assumed that people using the old `Cluster` constructor would not rely on the `internalTopics()` method so it didn't matter. I guess you're thinking of an older kafka-streams with a newer kafka-clients? We use private API within Kafka projects, so I don't think that would work. What do you think?